spynnaker/pyNN/models/neuron/population_machine_synapses.py

```diff
@@ -266,3 +268,12 @@ def max_gen_data(self) -> int:
     @overrides(AbstractSynapseExpandable.bit_field_size)
     def bit_field_size(self) -> int:
         return self._synaptic_matrices.bit_field_size
+
+    @overrides(AbstractReceiveRegionsToHost.get_download_regions)
+    def get_download_regions(
+            self, placement: Placement) -> Sequence[Tuple[int, int, int]]:
+
+        # Make sure not to overwrite the recording regions
+        start_index = len(self._pop_vertex.get_recording_variables())
+        return self._synaptic_matrices.get_download_regions(
```

Comment:
The local_recording_index for recording variables is fixed regardless which ones are actually recordings.
For example V is always 0 and Spikes always 3

So you can not use len of recording variables!

Reply:
fixed by using get_recordable_variables